molgenis-js/src/main/java/org/molgenis/js/magma/JsMagmaScriptEvaluator.java

```diff
@@ -79,6 +124,38 @@ public Object eval(String expression, Entity entity, int depth)
 		return value;
 	}
 
+	/**
+	 * Creates magmascript bindings for a given Entity with default depth.
+	 *
+	 * @param entity the entity to bind to the magmascript $ function
+	 * @return Bindings with $ function bound to the entity
+	 */
+	public Bindings createBindings(Entity entity)
+	{
+		return createBindings(entity, ENTITY_REFERENCE_DEFAULT_FETCHING_DEPTH);
+	}
+
+	/**
+	 * Creates magmascript bindings for a given Entity.
+	 *
+	 * @param entity the entity to bind to the magmascript $ function
+	 * @param depth  maximum depth to follow references when creating the entity value map
+	 * @return Bindings with $ function bound to the entity
+	 */
+	public Bindings createBindings(Entity entity, int depth)
+	{
+		Bindings bindings = new SimpleBindings();
```

Comment:
Does this conform with:

> You can share a ScriptEngine and CompiledScript objects across threads. They are threadsafe. Actually, you should share them, as a single engine instance is a holder for a class cache and for JavaScript objects' hidden classes, so by having only one you cut down on repeated compilation.

> What you can't share is Bindings objects. The bindings object basically corresponds to the JavaScript runtime environment's Global object. The engine starts with a default bindings instance, but if you use it in multithreaded environment, you need to use engine.createBindings() to obtain a separate Bindings object for every thread -- its own global, and evaluate the compiled scripts into it. That way you'll set up isolated global scopes with the same code.

It seems like bindings content is shared between script executions?

Reply:
The bindings shouldn't be shared between threads. They are supposed to be shortlived and used only for a single set of script executions on the entity.

Reply:
I've switched to evalList method to hide the details, so they won't be shared for sure now.